src/Lights/Shadows/shadowGenerator.ts

```diff
@@ -656,10 +681,100 @@ export class ShadowGenerator implements IShadowGenerator {
         if (this._shadowMap2) {
             return this._shadowMap2;
         }
-
         return this._shadowMap;

```

Comment:
not sure if this should also handle CSM

Reply:
No idea what shadowMap2 was used for, so did not touch it. Can always KIV.

Reply:
This is the one used by the materials as it might come from a post process in case of blur applied to the map for instance.

We need to handle it in case pp are applied to the shadows.

Reply:
ah, I see it now. ok fixed by adding ```private _cascadeShadowMaps2: Array<RenderTargetTexture>;``` and
```
public getCSMForRendering(index: number): RenderTargetTexture {
     if (this._cascadeShadowMaps2[index]) {
          return this._cascadeShadowMaps2[index];
     }
     return this._cascadeShadowMaps[index];
}
```
and in the onAfterUnbindObservable, fixed to
```let shadowMap = this.getCSMForRendering(index);```